tools/test.py

```diff
@@ -127,10 +131,25 @@
                                "Currently set search path: %s"
                        % (toolchain, search_path))
 
+         # boolean, true if test configuration file is for module false if mbed-os interface
+        module_conf = False
+
+        # Assign config file. Precedence: test_config>app_config
+        # TODO: merge configs if both given
+        if options.test_config:
+            config, module_conf = get_test_config(options.test_config, mcu)
+            if not config:
+                args_error(parser, "argument --test-config contains invalid path or identifier")
+        elif not options.app_config:
+            config = TestConfig.get_default_config(mcu)
+        else:
+            config = options.app_config
```

Comment:
@theotherjimmy would it be possible to allow the tools to accept a dictionary rather than a json file? Otherwise, I'd need to write a new file or change the existing one.